slm_porosity/heat_transfer.f90

```diff
@@ -458,8 +455,22 @@ SUBROUTINE user_stats ( u ,j_mn, startup_flag)
     INTEGER , INTENT (IN) :: startup_flag
     INTEGER , INTENT (IN) :: j_mn
     REAL (pr), DIMENSION (nwlt,1:n_var), INTENT (IN) :: u
-  END SUBROUTINE user_stats
+    REAL (pr) :: liquid_sum
+    REAL(pr), DIMENSION(1) :: width, length, depth
 
+    IF (startup_flag.EQ.0) THEN
+      OPEN(12, file="liquid_fraction.txt", status="replace", action="write")
+      CLOSE(12)
+    ELSE
+      liquid_sum = SUM(u(:,n_var_lfrac)*dA(:))
+      length = x(MAXLOC(ABS(u(:,n_var_lfrac)*x(:,1))), 1)
```

Comment:
should be some difference of coordinates

Reply:
What you mean about difference?

Reply:
L = x_2 - x_1

Reply:
That's right. But I can implement it only with cycle and lot of variables. Middle school like programming.

Reply:
First, it is not excuse for the wrong code!
Second, it is very easy. L = x_max - x_min.  What is the problem?

Reply:
If all of two points are above zero, then equating them on lf gives some range (x1,x2) where x2 a priori is bigger. MIN would give some zero point, MAX would give x2 point. So there must be some cycle to find x1. The problem is in some sort of ugliness of the solution for that, which I have.